lib/tessel/deployment/rust.js

```diff
@@ -28,7 +28,7 @@ var exportables = {
       var cargoToml = toml.parse(fs.readFileSync(path.join(pushdir, 'Cargo.toml'), 'utf8'));
 
       if (cargoToml.package) {
-        basename = path.basename(program);
+        basename = pushdir;
```

Comment:
The "package" field is in our own rust init generated Cargo.toml... IIRC, this path is what allows for deployment from the root of a project, but where the program code is in a subdirectory, eg. `t2 run src/whatever.rs`. If that's correct, then the change here would break that—someone should smoke test. Also, this change should have its own tests.